posts/2023-05-01-wasi-target-rename-1.71.0.md

```diff
@@ -0,0 +1,47 @@
+---
+layout: post
+title: "Renaming the WASI target in Rust 1.71.0"
+author: Yoshua Wuyts
+---
+In Rust 1.71 the existing `wasm32-wasi` target will be renamed
+to `wasm32-wasi-preview1`. This matches the naming introduced in the [WASI
+Preview Framework][wpf], whose scheme has already been adopted by other
+language toolchains such as [Golang] and [Node.js]. It is expected that
+eventually WASI will eventually stabilize far enough that it will reach a 1.0
+status. By freeing up the `wasm32-wasi` target name now, it enables us to
+reintroduce it in the future as the target name for an eventual 1.0 stable WASI
+release.
+
+While we don’t have any immediate plans to deprecate the Preview 1 target, the
+reality is that most of the development of WASI in recent years has centered
+around WASI Preview 2 and [Wasm Components][components]. The expectation is that
+Preview 2 will eventually supplant Preview 1, and support across the WASI
+ecosystem will gradually wind down for Preview 1. By adopting “preview” in the
```

Comment:
> By adopting “preview” in the target names

Note: "preview" isn't actually used in the target names just "p" which stands for preview.